### ISSUE TITLE
Lux example MLP with TrainState API throws JIT session errors

### ISSUE THREAD

**Issue by mashu** [EnzymeAD/Reactant.jl#220]
This is fresh project create with `]generate mwe` and adding only packages used in example from docs.

Project.toml is
```text
name = "mwe"
uuid = "cdc3416e-61dc-46b3-bdd4-a6c718dbbe83"
authors = ["Mateusz Kaduk <mateusz.kaduk@gmail.com>"]
version = "0.1.0"

[deps]
Enzyme = "7da242da-08ed-463a-9acd-ee780be4f1d9"
Functors = "d9f16b24-f501-4c13-a1f2-28368ffc5196"
Lux = "b2108857-7c20-44ae-9111-449ecde12c47"
Optimisers = "3bd65402-5787-11e9-1adc-39752487f4e2"
Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
Reactant = "3c362404-f566-11ee-1572-e11a4b42c853"
Zygote = "e88e6eb3-aa80-5325-afca-941959d7151f"

[compat]
Enzyme = "0.13.13"
Functors = "0.4.12"
Lux = "1.2.0"
Optimisers = "0.3.3"
Printf = "1.11.0"
Random = "1.11.0"
Reactant = "0.2.3"
Zygote = "0.6.72"
```
The code is taken from Lux examples
```julia
using Lux, Reactant, Enzyme, Random, Zygote
using Functors, Optimisers, Printf
Reactant.set_default_backend("gpu")
const xdev = xla_device()

model = Chain(
    Dense(2 => 4, gelu),
    Dense(4 => 4, gelu),
    Dense(4 => 2)
)
ps, st = Lux.setup(Random.default_rng(), model)

x_ra = [randn(Float32, 2, 32) for _ in 1:32]
y_ra = [xᵢ .^ 2 for xᵢ in x_ra]
ps_ra = ps |> xdev
st_ra = st |> xdev

dataloader = DeviceIterator(xdev, zip(x_ra, y_ra))

function train_model(model, ps, st, dataloader)
    train_state = Training.TrainState(model, ps, st, Adam(0.001f0))

    for iteration in 1:1000
        for (xᵢ, yᵢ) in dataloader
            grads, loss, stats, train_state = Training.single_train_step!(
                AutoEnzyme(), MSELoss(), (xᵢ, yᵢ), train_state)
        end
        if iteration % 100 == 0 || iteration == 1
            # We need to do this since scalar outputs are currently expressed as a zero-dim
            # array
            loss = Array(loss)[]
            @printf("Iter: [%4d/%4d]\tLoss: %.8f\n", iteration, 1000, loss)
        end
    end

    return train_state
end

train_model(model, ps_ra, st_ra, dataloader)
```

Only `Reactant.set_default_backend("gpu")` was added but error happens without it. When setting GPU backend additionally I also get warrning about scalar indexing. 

The errors and warning look like this
```
JIT session error: Symbols not found: [ ArrayFromHostBuffer, BufferOnCPU, BufferToClient, BufferToDevice, BufferToHost, ClientGetDevice, FutureAwait, UnsafeBufferPointer, pcre2_match_data_free_8, pcre2_match_data_create_from_pattern_8, pcre2_match_context_create_8, pcre2_match_8, pcre2_jit_stack_create_8, pcre2_jit_stack_assign_8, pcre2_jit_compile_8, pcre2_get_ovector_pointer_8, pcre2_get_error_message_8, pcre2_compile_8 ]
JIT session error: Symbols not found: [ ArrayFromHostBuffer, BufferOnCPU, BufferToClient, BufferToDevice, BufferToHost, ClientGetDevice, FutureAwait, UnsafeBufferPointer, pcre2_match_data_free_8, pcre2_match_data_create_from_pattern_8, pcre2_match_context_create_8, pcre2_match_8, pcre2_jit_stack_create_8, pcre2_jit_stack_assign_8, pcre2_jit_compile_8, pcre2_get_ovector_pointer_8, pcre2_get_error_message_8, pcre2_compile_8 ]
JIT session error: Symbols not found: [ ArrayFromHostBuffer, BufferOnCPU, BufferToClient, BufferToDevice, BufferToHost, ClientGetDevice, FutureAwait, UnsafeBufferPointer, pcre2_match_data_free_8, pcre2_match_data_create_from_pattern_8, pcre2_match_context_create_8, pcre2_match_8, pcre2_jit_stack_create_8, pcre2_jit_stack_assign_8, pcre2_jit_compile_8, pcre2_get_ovector_pointer_8, pcre2_get_error_message_8, pcre2_compile_8 ]
JIT session error: Symbols not found: [ ArrayFromHostBuffer, BufferOnCPU, BufferToClient, BufferToDevice, BufferToHost, ClientGetDevice, FutureAwait, UnsafeBufferPointer, pcre2_match_data_free_8, pcre2_match_data_create_from_pattern_8, pcre2_match_context_create_8, pcre2_match_8, pcre2_jit_stack_create_8, pcre2_jit_stack_assign_8, pcre2_jit_compile_8, pcre2_get_ovector_pointer_8, pcre2_get_error_message_8, pcre2_compile_8 ]
```
and
```
┌ Warning: Performing scalar get-indexing on task Task (runnable, started) @0x00007f2ee6cb6a40.
│ Invocation resulted in scalar indexing of a ConcreteRArray.
│ This is typically caused by calling an iterating implementation of a method.
│ Such implementations *do not* execute on device, but very slowly on the CPU,
│ and require expensive copies and synchronization each time and therefore should be avoided.
└ @ Reactant ~/.julia/packages/Reactant/rRa4g/src/ConcreteRArray.jl:144
Iter: [   1/1000]       Loss: 2.69598722
```

**Comment by avik-pal** [comment]
@wsmoses do you know where the JIT session errors are from?

> When setting GPU backend additionally I also get warrning about scalar indexing.

This one is coming from `loss = Array(loss)[]`. Don't worry about it for now, once we release the latest version of Reactant which has `ConcreteRNumber`, we can make it go away.

**Comment by wsmoses** [comment]
This is the part of Enzyme.jl 1.11 support I'm working on right now ...... which implies that reactant compile isn't being used

**Comment by avik-pal** [comment]
> which implies that reactant compile isn't being used

This is what I thought, but the function being called is Reactant Thunk object

```julia
ts.cache.extras.compiled_grad_and_step_function = Reactant.Compiler.Thunk{Symbol("##compute_gradients_internal_and_step!_reactant#517153")}()
```

**Comment by wsmoses** [comment]
Try on 1.10 in the interim that shouldn't have the jit errors and hopefully give better debug info.

**Comment by avik-pal** [comment]
fixed with enzyme 0.13.14
